src/web-server.js

```diff
@@ -332,6 +332,75 @@ app.delete("/services/:serviceName", function (req, res) {
     });
 });
 
+app.delete("/scopes", function (req, res) {
+  let body;
+  if (typeof req.body === "string") {
+    try {
+      body = JSON.parse(req.body);
+    } catch (e) {
+      return res.status(400).send("Invalid request body");
+    }
+  } else {
+    body = req.body;
+  }
+
+  if (body.regexString === undefined) {
+    return res
+      .status(400)
+      .send("The regex key in the request body must have a value");
+  }
+  if (typeof body.regexString !== "string") {
+    return res.status(400).send("The regex value must be a string");
+  }
+  if (body.regexString.length === 0) {
+    return res.status(400).send("The regex value must not be an empty string");
+  }
+
+  let regex;
+
+  try {
+    regex = new RegExp(body.regexString);
+  } catch (e) {
+    return res.status(400).send("The regex value is not valid regex");
+  }
+
+  const env = getEnv(req);
+  const isDryRun = req.query.dry_run === "true" || req.query.dry_run === "";
+
+  if (isDryRun) {
+    return ioOperations
+      .readManifest(env)
+      .then((data) => {
+        let scopesToBeDeleted = Object.fromEntries(
```

Comment:
`scopesToBeDeleted` should be calculated in the same place in the code, to avoid the endpoint behaving differently depending on whether the dry run query parameter is present. Creating a function within modify.js to calculate `scopesToBeDeleted` would be best, so it can be reused within `deleteScopes` and by the code here.